File: 3291-find-if-array-can-be-sorted/3291-find-if-array-can-be-sorted.c
bool canSortArray(int* nums, int numsSize) {
    int prev = -1 ; 
    for(int i = 0 ; i<numsSize ; ){
        int max = -1 ; 
        int min = 2e8+1 ;
        int temp = __builtin_popcount(nums[i]) ;
        while(i<numsSize&&__builtin_popcount(nums[i])==temp){
            if(nums[i]>max) max = nums[i] ;
            if(nums[i]<min)  min = nums[i] ; 
            i++ ; 
        }
        if(min<prev) return 0 ; 
        prev = max ;
    }
    return 1;
}

File: 3291-find-if-array-can-be-sorted/3291-find-if-array-can-be-sorted.java
class Solution {
    public int get(int n){
        return Integer.bitCount(n);
    }
    public boolean canSortArray(int[] nums) {
        int prev = Integer.MIN_VALUE ; 
        for(int i = 0 ;i<nums.length ;){
            int max = -1 ;
            int min = Integer.MAX_VALUE ; 
            int temp = get(nums[i]);
            while(i<nums.length&&get(nums[i])==temp){
                if(nums[i]>max){
                    max = nums[i] ;
                }
                if(nums[i]<min){
                    min = nums[i] ; 
                }
                i++ ; 
            }
            if(min<prev){
                return false ;
            }
            prev = max ;
        }
        return true ;
    }
}